src/System.Text.Primitives/System/Text/FormattingData.cs

```diff
@@ -49,6 +339,7 @@ static FormattingData()
                 new byte[] { 43, 0, }, // plus sign 
                 new byte[] { 78, 0, 97, 0, 78, 0, }, // NaN
                 new byte[] { 69, 0, }, // E
+                new byte[] { 101, 0, }, // e
```

Comment:
If you are adding a new symbol, you need to also change System.Text.Globalization, I think.